File: 프로그래머스/1/134240. 푸드 파이트 대회/푸드 파이트 대회.js
function solution(foods) {
    let answer = "";
    for (let i = 1; i < foods.length; i++) {
        const qty = Math.floor(foods[i] / 2);
        answer += (i+'').repeat(qty);
    }
    return answer + "0" + [...answer].reverse().join("");
}

File: 프로그래머스/1/134240. 푸드 파이트 대회/푸드 파이트 대회.java
class Solution {
    public String solution(int[] food) {
        StringBuilder sb = new StringBuilder();
        
        for (int i = 1; i < food.length; i++) {
            int cnt = food[i] / 2;
            sb.append(String.valueOf(i).repeat(cnt));
        }
        
        String reversed = new StringBuilder(sb).reverse().toString();
        return sb.append("0").append(reversed).toString();
    }
}